### PR TITLE
Add clearing of filtered_sensors after each loop

### PR DESCRIPTION
**Pull request by abihall**
**Add clearing of filtered_sensors after each loop in sensor filter**
filtered sensors was not cleared after each loop through the sensors when resetting the sensor filter. This makes sure it is cleared after using it to calculate forward kinematics and averages on a iteration

### CODE REVIEW

**Review by Bidski — DISMISSED** [review]
Seems important

**Comment by Bidski** [comment]
The alternative would be to move the declaration of `filtered_sensors` inside the for loop since that is the only place it is used

**Comment by KipHamiltons** [comment]
I suppose it becomes a question of whether it's worth keeping that in the scope it's used vs incurring a heap allocation every iteration

**Review by tayloryoung6396 — COMMENTED** [review]
I think if you're not using any of the information between loops you're better clearing the entire variable or redeclaring it each loop, stops people making stupid mistakes in the future

**Comment by KipHamiltons** [comment]
> I think if you're not using any of the information between loops you're better clearing the entire variable or redeclaring it each loop, stops people making stupid mistakes in the future

Good point. That does stupid-proof it a bit :laughing:

File: module/input/SensorFilter/src/SensorFilter.cpp [hunk 1]
```diff
@@ -196,16 +196,17 @@ namespace module::input {
...
-                    Eigen::Vector3d gyro  = Eigen::Vector3d::Zero();
-                    Eigen::Vector3d rMFt  = Eigen::Vector3d::Zero();
-                    auto filtered_sensors = std::make_unique<Sensors>();
+                    Eigen::Vector3d acc  = Eigen::Vector3d::Zero();
+                    Eigen::Vector3d gyro = Eigen::Vector3d::Zero();
+                    Eigen::Vector3d rMFt = Eigen::Vector3d::Zero();
 
                     for (const auto& s : sensors) {
+
+                        auto filtered_sensors = std::make_unique<Sensors>();
```
Comment (TrentHouliston) [inline module/input/SensorFilter/src/SensorFilter.cpp:205]: No need for this to be a unique_ptr, just make it a stack variable

**Review by TrentHouliston — COMMENTED** [review]
Another comment about this, you have to be careful about using the sensors like this. Sometimes you'll get errors on a servo in a packet and since you're using raw sensors directly, you'll end up with some undefined values in your IK.
This could make some very bad things happen to your averages

**Review by KipHamiltons — DISMISSED** [review]
LGTM. Can follow up with error handling for the servos :+1:

File: module/input/SensorFilter/src/SensorFilter.cpp [hunk 2]
```diff
@@ -196,48 +196,48 @@ namespace module::input {
...
-                    Eigen::Vector3d acc   = Eigen::Vector3d::Zero();
-                    Eigen::Vector3d gyro  = Eigen::Vector3d::Zero();
-                    Eigen::Vector3d rMFt  = Eigen::Vector3d::Zero();
-                    auto filtered_sensors = std::make_unique<Sensors>();
+                    Eigen::Vector3d acc  = Eigen::Vector3d::Zero();
+                    Eigen::Vector3d gyro = Eigen::Vector3d::Zero();
+                    Eigen::Vector3d rMFt = Eigen::Vector3d::Zero();
 
                     for (const auto& s : sensors) {
+                        auto filtered_sensors = Sensors();
```
Comment (TrentHouliston) [inline module/input/SensorFilter/src/SensorFilter.cpp:204]:
```suggestion
                        Sensors filtered_sensors;
```

**Review by TrentHouliston — CHANGES_REQUESTED** [review]

File: module/input/SensorFilter/src/SensorFilter.cpp (hunk 2 above)
Comment (KipHamiltons) [reply module/input/SensorFilter/src/SensorFilter.cpp:204]: I mean, at least call the ctor explicitly :p 
```suggestion
                        Sensors filtered_sensors{};
```

Comment (abihall) [reply module/input/SensorFilter/src/SensorFilter.cpp:204]: difference being?

**Review by TrentHouliston — APPROVED** [review]

**Pull request merged**